int8 vin
float32[] data